packages/jwt/src/jwt-validator.ts

```diff
@@ -0,0 +1,234 @@
+            // Check for duplicate scopes within each permission
+            if (perm.scope) {
+              const scopes = new Set<string>();
+              for (const scope of perm.scope) {
+                if (scopes.has(scope)) {
+                  result.errors.push("Duplicate scope in permission");
+                  valid = false;
+                  continue;
+                }
+                scopes.add(scope);
+              }
+            }
+
+            // Check for duplicate services and validate deployment dependencies
+            if (perm.deployments) {
+              for (const deployment of perm.deployments) {
+                if (!this.validateDeployment(deployment, result)) {
+                  valid = false;
+                }
+              }
+            }
+          }
+        }
+      } else if (payload.leases?.access === "full" && payload.leases?.permissions) {
+        result.errors.push("Permissions not allowed for full access");
+        valid = false;
+      }
+
+      result.isValid = result.errors.length === 0;
+    } catch (error) {
+      const errorMessage = error instanceof Error ? error.message : String(error);
+      result.errors.push(`Error during JWT validation: ${errorMessage}`);
+      // Optionally log the error for debugging
+      console.error("JWT validation error:", error);
+    }
+
+    return result;
+  }
+
+  /**
+   * Validates deployment structure and dependencies
+   * @param deployment The deployment to validate
+   * @param result The validation result to update
+   * @returns Whether the validation passed
+   */
+  private validateDeployment(deployment: any, result: JwtValidationResult): boolean {
+    let valid = true;
+
+    // Check for duplicate scopes within deployment
+    const scopes = new Set<string>();
+    for (const scope of deployment.scope) {
+      if (scopes.has(scope)) {
+        result.errors.push("Duplicate scope in deployment");
+        valid = false;
+        break;
+      }
+      scopes.add(scope);
+    }
+
+    // Validate deployment dependencies
+    if (deployment.gseq && !deployment.dseq) {
+      result.errors.push("gseq requires dseq");
+      valid = false;
+    }
+    if (deployment.oseq && (!deployment.dseq || !deployment.gseq)) {
+      result.errors.push("oseq requires dseq and gseq");
+      valid = false;
+    }
+    if (deployment.dseq && !deployment.services) {
+      result.errors.push("services required when dseq is present");
+      valid = false;
+    }
+    if (deployment.services && !deployment.dseq) {
+      result.errors.push("services requires dseq");
+      valid = false;
+    }
+
+    // Check for duplicate services
+    if (deployment.services) {
+      const services = new Set<string>();
+      for (const service of deployment.services) {
+        if (services.has(service)) {
+          result.errors.push("Duplicate service in deployment");
+          valid = false;
+          break;
+        }
+        services.add(service);
+      }
```

Comment:
thought(blocking): this code is easily extractable to a function and can be then reused between calls:

```ts
const duplicatedServices = findDuplicated(deployment.services));
duplicatedServices.forEach(service => result.errors.push(`Duplicated service in deployment: ${service}`));
```
Or even this string could be passed as a parameter to findDuplicated, so then you could do:

```ts
result.errors.push(
  ...collectDuplicated(deployment.services, service => `Duplicated service in deployment: ${service}`)
)
```